src/main.rs

```diff
@@ -71,7 +71,13 @@ fn setup(config_folder: &Path) -> IoResult<(Data<RwLock<Config>>, ServerConfig,
     // Create necessary directories.
     fs::create_dir_all(&server_config.upload_path)?;
     for paste_type in &[PasteType::Url, PasteType::Oneshot, PasteType::OneshotUrl] {
-        fs::create_dir_all(paste_type.get_path(&server_config.upload_path))?;
+        let _ = paste_type
+            .get_path(&server_config.upload_path)
+            .ok_or(IoError::new(
+                IoErrorKind::Other,
+                String::from("Invalid upload path"),
+            ))
+            .map(fs::create_dir_all)?;
```

Comment:
I don't think the error from `create_dir_all` is handled here. I simulated some error with the following snippet:

```rust
let _ = paste_type
    .get_path(&server_config.upload_path)
    .ok_or(IoError::new(
        IoErrorKind::Other,
        String::from("Invalid upload path"),
    ))
    .map(|v| {
        let result = fs::create_dir_all(v);
        IoError::new(IoErrorKind::Other, String::from("some error happened"))
    })?;
```

And didn't see any errors in the application.

Instead, you can probably do this:

```rs
let upload_path = paste_type
    .get_path(&server_config.upload_path)
    .ok_or(IoError::new(
        IoErrorKind::Other,
        String::from("Invalid upload path"),
    ))?;
fs::create_dir_all(upload_path)?;
```

Reply:
Yep, you're absolutely right. It turns out `and` is what I want. But it's much easier to use 2 separate statements.